lib/travis/build/git/clone.rb

```diff
@@ -17,7 +17,17 @@ def apply
 
           def clone_or_fetch
             sh.if "! -d #{dir}/.git" do
-              sh.cmd "git clone #{clone_args} #{data.source_url} #{dir}", assert: true, retry: true
+              if sparseCheckout?
+                sh.cmd "git init #{dir}", assert: true, retry: true
+                sh.cmd "git -C #{dir} git config core.sparseCheckout true", assert: true, retry: true
+                sh.cmd "echo #{sparseCheckout} >> #{dir}/.git/info/sparseCheckout", assert: true, retry: true
```

Comment:
I don't think this is going to work, because the file specified in the `sparseCheckout` (keeping the current option name) is not available to `git` before actually checking out the code.

The entire content of `.git/info/sparseCheckout` should be specified as a string instead.